### PR TITLE
English Dictionary Homework

### PR DESCRIPTION
**Pull request by AlizaBuchen**
EnglishDictionary

### CODE REVIEW

File: src/test/java/buchen/dictionary/EnglishDictionaryTest.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+class EnglishDictionaryTest {
+
+    @Test
+    public void getDefinition() throws CsvValidationException, IOException {
+        // given
+        EnglishDictionary dictionary = new EnglishDictionary();
+
+        // when
+        List<String> definitions = dictionary.getDefinition("slee");
```
Comment (andrewoid) [inline src/test/java/buchen/dictionary/EnglishDictionaryTest.java]: Test a word with multiple meanings

File: src/test/java/buchen/dictionary/EnglishDictionaryTest.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+        // then
+        assertEquals("[[To slay.]]", definitions.toString());
```
Comment (andrewoid) [inline src/test/java/buchen/dictionary/EnglishDictionaryTest.java]: Assert each item in the list

File: src/main/java/buchen/dictionary/DictionaryFrame.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+
+        defArea = new JTextArea();
+        defArea.setEditable(false);
+        defArea.setLineWrap(true);
+        defArea.setWrapStyleWord(true);
+
+
+        add(wordField, BorderLayout.NORTH);
+        add(defArea);
```
Comment (andrewoid) [inline src/main/java/buchen/dictionary/DictionaryFrame.java]: Which region is this being added into?

**Review by andrewoid — COMMENTED** [review]
Add a readme

File: src/main/java/buchen/dictionary/EnglishDictionary.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    /**
+     * @param word to look up.
+     * @return a List of definitions, or an empty list if the word doesn't exist.
+     */
+    public List<String> getDefinition(String word) {
+        List<String> def = new ArrayList<>();
+
+        for (String[] record : words) {
+            if (record[0].equalsIgnoreCase(word)) {
+                def.add(Arrays.toString(record[2].split(";")));
```
Comment (andrewoid) [inline src/main/java/buchen/dictionary/EnglishDictionary.java]: Why are you doing this?

File: src/main/java/buchen/dictionary/DictionaryFrame.java [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+    }
+
+    private void updateDefinition() {
+        SwingUtilities.invokeLater(() -> {
+            String word = wordField.getText();
+            List<String> defs = engDic.getDefinition(word);
+            defArea.setText("");
+            for (String def : defs) {
+                defArea.append(def + "\n");
```
Comment (andrewoid) [inline src/main/java/buchen/dictionary/DictionaryFrame.java]: `defArea.append(def).append("\n");`

Comment (AlizaBuchen) [reply src/main/java/buchen/dictionary/DictionaryFrame.java]: Error


When I write it this way It's giving me an error, do you have any advice?

Comment (andrewoid) [reply src/main/java/buchen/dictionary/DictionaryFrame.java]: oops.
```
defArea.append(def);
defArea.append("\n");
```

**Review by andrewoid — APPROVED** [review]

**Pull request merged**